vocabularies/Common.xml

```diff
@@ -1450,6 +1450,68 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
+        <Annotation Term="Core.AppliesViaContainer" />
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="AI-based recommendations for entities in the annotation target" />
+      </Term>
+      <ComplexType Name="RecommendationsType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="Action" Type="Common.NamespaceQualifiedActionName">
+          <Annotation Term="Core.Description" String="Name of an action (or function) that retrieves recommendations (see [template](#Template_RecommendationsAction))" />
+        </Property>
+        <Property Name="Dependencies" Type="Collection(Common.RecommendationsDependency)">
+          <Annotation Term="Core.Description" String="Predictors for each recommendation target" />
+        </Property>
+      </ComplexType>
+      <Action Name="Template_RecommendationsAction" IsBound="true">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Template for actions that retrieve AI-based recommendations for the bound entity and its related entities" />
+        <Annotation Term="Core.LongDescription">
+          <String>The template action itself cannot be invoked.</String>
+        </Annotation>
+        <Parameter Name="Entity" Type="Edm.EntityType" Nullable="false">
+          <Annotation Term="Core.Description" String="Entity relative to which the `EntityKeys` are interpreted" />
+        </Parameter>
+        <Parameter Name="EntityKeys" Type="Edm.EntityType" Nullable="false">
+          <Annotation Term="Core.Description" String="Nested structure containing the keys of entities for which recommendations shall be retrieved" />
+          <Annotation Term="Core.LongDescription">
+            <String>This parameter has the same structure as the binding parameter
+            and its related entity types, but contains only their key properties.
+            If this parameter does not exceed the URL size, a function can be used instead of an action.</String>
+          </Annotation>
+        </Parameter>
+        <ReturnType Type="Collection(Common.PropertyRecommendationType)" Nullable="false" />
+      </Action>
+      <ComplexType Name="PropertyRecommendationType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="target" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="Target for this recommendation, format tbd" />
+        </Property>
+        <Property Name="value" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="Recommended value, converted to string" />
+        </Property>
+        <Property Name="text" Type="Edm.String" Nullable="true">
+          <Annotation Term="Core.Description" String="Description of the recommended value" />
+        </Property>
+        <Property Name="alternatives" Type="Collection(Common.AlternativeRecommendationType)" Nullable="false">
```

Comment:
Can the alternatives also appear as items of the returned collection?

Reply:
There is _one_ recommendation and zero or more alternatives in the return value. What do you mean by "returned collection"?

Reply:
Line 1489 reads
~~~xml
 <ReturnType Type="Collection(Common.PropertyRecommendationType)" Nullable="false" />
~~~

Should the "Collection" be removed?

Reply:
> Should the "Collection" be removed?

No, there is one `PropertyRecommendationType` per targeted property.